Content.Server/Doors/Systems/DoorSystem.cs

```diff
@@ -284,9 +284,15 @@ private void OnEmagged(EntityUid uid, DoorComponent door, GotEmaggedEvent args)
         {
             if (door.State == DoorState.Closed)
             {
-                StartOpening(uid);
-                airlockComponent?.SetBoltsWithAudio(!airlockComponent.IsBolted());
-                args.Handled = true;
+                SetState(uid, DoorState.Emagging, door);
+                PlaySound(uid, door.SparkSound.GetSound(), AudioParams.Default.WithVolume(8), args.UserUid, false);
+
+                door.Owner.SpawnTimer(door.EmagDuration, () =>
+                {
+                    StartOpening(uid);
+                    airlockComponent?.SetBoltsWithAudio(!airlockComponent.IsBolted());
+                    args.Handled = true;
+                });
```

Comment:
Please don't use timers.

Reply:
Sorry if I'm just a little out of the loop on this one, is there an alternative or do you just mean no timer at all?

Reply:
Systems should track time themselves and timercomponent should be avoided.

Reply:
so yeah, it does look like shareddoorsystem actually is actually keeping track of the time between door states, and on change it raises the DoorStateChangedEvent.

I couldn't actually find a way to tap into when the specific state so I have it including the previous state that it swapped from as well.

If this isn't a good way of going about it just let me know and I can keep brain storming.